app/web/service/events.ts

```diff
@@ -249,19 +249,40 @@ export async function listAllEvents({
   return res.toObject();
 }
 
-interface ListMyEventsInput {
+export interface ListMyEventsInput {
+  pageNumber?: number;
   pageSize?: number;
   pageToken?: string;
+  pastEvents?: boolean;
+  showCancelled?: boolean;
 }
 
-export async function listMyEvents({ pageSize, pageToken }: ListMyEventsInput) {
+export async function listMyEvents({
+  pageNumber,
+  pageSize,
+  pageToken,
+  pastEvents,
+  showCancelled,
+}: ListMyEventsInput) {
   const req = new ListMyEventsReq();
+  req.setAttending(true);
+  req.setOrganizing(true);
+
+  if (pastEvents !== undefined) {
+    req.setPast(pastEvents);
+  }
+  if (pageNumber) {
+    req.setPageNumber(pageNumber);
+  }
   if (pageSize) {
     req.setPageSize(pageSize);
   }
   if (pageToken) {
     req.setPageToken(pageToken);
   }
+  if (showCancelled !== undefined) {
```

Comment:
`if (showCancelled)` = `if (showCancelled !== undefined)` since it's a boolean?

Reply:
Then it won't set the showCancelled value in the request if it's `false`...

Reply:
🤦‍♂️👍